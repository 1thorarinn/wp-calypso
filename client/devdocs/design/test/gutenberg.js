/**
 * External dependencies
 */
import React from 'react';
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import DocumentHead from 'components/data/document-head';
import Gutenberg from '../gutenberg';
import Main from 'components/main';
import ReadmeViewer from 'components/readme-viewer';

describe( 'Gutenberg', () => {
	test( 'should render', () => {
		const wrapper = shallow( <Gutenberg/> );

		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should have a main component', () => {
		const wrapper = shallow( <Gutenberg/> );
		const main = wrapper.find( Main );

		expect( main ).toHaveLength( 1 );
		expect( main.hasClass( 'devdocs devdocs__gutenberg' ) ).toBe( true );
	} );

	test( 'should have a document head component', () => {
		const wrapper = shallow( <Gutenberg/> );
		const documentHead = wrapper.find( DocumentHead );

		expect( documentHead ).toHaveLength( 1 );
		expect( documentHead.props().title ).toBe( 'Gutenberg Blocks' );
	} );

	test( 'should have a readme viewer component', () => {
		const wrapper = shallow( <Gutenberg/> );
		const readmeViewer = wrapper.find( ReadmeViewer );

		expect( readmeViewer ).toHaveLength( 1 );
		expect( readmeViewer.props().readmeFilePath ).toBe( '/client/devdocs/gutenberg/README.md' );
	} );
} );
