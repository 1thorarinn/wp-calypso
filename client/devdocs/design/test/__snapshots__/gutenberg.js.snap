// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gutenberg should render 1`] = `
<Main
  className="devdocs devdocs__gutenberg"
>
  <Connect(DocumentHead)
    title="Gutenberg Blocks"
  />
  <ReadmeViewer
    readmeFilePath="/client/devdocs/gutenberg/README.md"
  />
</Main>
`;
